lib/dependencies/SystemPlugin.js

```diff
@@ -39,11 +43,35 @@ class SystemPlugin {
 						parser.state.module.context, require.resolve("../../buildin/system.js"));
 					return ParserHelpers.addParsedVariableToModule(parser, "System", systemPolyfillRequire);
 				});
+
+				parser.hooks.call.for("System.import").tap("SystemPlugin", expr => {
+					if(shouldWarn) {
+						compilation.warnings.push(new SystemImportDeprecationWarning(parser.state.module, expr.loc));
```

Comment:
Still not happy with this. To make the warning cache correctly, it has to be in `module.warnings`, but I will try changing it myself.